src/ports/linux/mib/lldpRemManAddrTable.c

```diff
@@ -122,19 +122,21 @@ netsnmp_variable_list * lldpRemManAddrTable_get_next_data_point (
    int error;
 
    iterator = (pf_port_iterator_t *)*my_loop_context;
-   port = pf_snmp_get_next_port (iterator);
-   if (port == 0)
-   {
-      return NULL;
-   }
 
-   error = pf_snmp_get_peer_timestamp (pnet, port, &timestamp);
-   if (error)
+   do
    {
-      return NULL;
-   }
-   error = pf_snmp_get_peer_management_address (pnet, port, &address);
-   if (error)
+      port = pf_snmp_get_next_port (iterator);
+      if (port != 0)
+      {
+         error = pf_snmp_get_peer_timestamp (pnet, port, &timestamp);
+      }
+      if (error == 0)
```

Comment:
This will run even if port == 0.
Perhaps return immediately when port == 0?

Reply:
Updated.
Done